policy-controller/k8s/index/src/index.rs

```diff
@@ -1428,6 +1454,39 @@ impl PolicyIndex {
         }
         true
     }
+
+    fn update_statuses(&self) {
+        // Construct a map where each key is a route, and its value is the
+        // list of Servers that accept it.
+        let accepted_routes: HashMap<String, Vec<String>> = self
+            .http_routes
+            .iter()
+            .map(|(route_name, route)| {
+                let servers = self
+                    .servers
+                    .keys()
+                    .filter_map(|server| {
+                        if route.selects_server(server) {
+                            Some(server.clone())
+                        } else {
+                            None
+                        }
+                    })
+                    .collect();
```

Comment:
nit, take it or leave it: might look slightly nicer to just write
```suggestion
                    .filter(|server| route.selects_server(server))
                    .cloned()
                    .collect();
```
but it's no big deal either way